### ISSUE TITLE
Wrong result column type for requested conversion when reading JSONB_ARRAYAGG in DB2

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14466]
This seems to be similar to a previous issue where the problem appeared with `JSONB_ARRAY` and `JSONB_OBJECT`: https://github.com/jOOQ/jOOQ/issues/13643

But this time, it appears in `JSONB_ARRAYAGG`, e.g. when running this:

```java
Record1<JSONB> r5 =
create().select(jsonbArrayAggDistinct(TBook_AUTHOR_ID()))
        .from(TBook())
        .fetchOne();
```

It produces this error:

```
org.jooq.exception.DataAccessException: SQL [select cast((('[' || listagg(distinct "DB2INST1"."T_BOOK"."AUTHOR_ID", ',')) || ']') as varchar(32672)) from "DB2INST1"."T_BOOK" -- SQL rendered with a free trial version of jOOQ 3.18.0-SNAPSHOT]; Error while reading field: json_arrayagg(distinct "DB2INST1"."T_BOOK"."AUTHOR_ID"), at JDBC index: 1
	at org.jooq_3.18.0-SNAPSHOT.DB2.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3406)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:746)
	at org.jooq.impl.CursorImpl$CursorIterator.fetchNext(CursorImpl.java:1465)
	at org.jooq.impl.CursorImpl$CursorIterator.hasNext(CursorImpl.java:1426)
	at org.jooq.impl.CursorImpl.fetchNext(CursorImpl.java:239)
	at org.jooq.impl.Tools.fetchOne(Tools.java:2678)
	at org.jooq.impl.ResultQueryTrait.fetchOne(ResultQueryTrait.java:509)
	at org.jooq.test.all.testcases.JSONTests.testJSONBArrayAggDistinct(JSONTests.java:669)
	at org.jooq.test.jOOQAbstractTest.testJSONBArrayAggDistinct(jOOQAbstractTest.java:10495)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
Caused by: java.sql.SQLException: Error while reading field: json_arrayagg(distinct "DB2INST1"."T_BOOK"."AUTHOR_ID"), at JDBC index: 1
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1591)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.apply(CursorImpl.java:1526)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.apply(CursorImpl.java:1)
	at org.jooq.impl.RecordDelegate.operate(RecordDelegate.java:144)
	at org.jooq.impl.CursorImpl$CursorIterator.fetchNext(CursorImpl.java:1450)
	... 38 more
Caused by: com.ibm.db2.jcc.am.SqlSyntaxErrorException: [jcc][t4][1092][11638][4.31.10] Invalid data conversion: Wrong result column type for requested conversion. ERRORCODE=-4461, SQLSTATE=42815
	at com.ibm.db2.jcc.am.b7.a(b7.java:810)
	at com.ibm.db2.jcc.am.b7.a(b7.java:66)
	at com.ibm.db2.jcc.am.b7.a(b7.java:133)
	at com.ibm.db2.jcc.am.bi.Q(bi.java:1939)
	at com.ibm.db2.jcc.am.ResultSet.getBytes(ResultSet.java:1270)
	at org.jooq.tools.jdbc.DefaultResultSet.getBytes(DefaultResultSet.java:170)
	at org.jooq.tools.jdbc.DefaultResultSet.getBytes(DefaultResultSet.java:170)
	at org.jooq.tools.jdbc.DefaultResultSet.getBytes(DefaultResultSet.java:170)
	at org.jooq.impl.CursorImpl$CursorResultSet.getBytes(CursorImpl.java:548)
	at org.jooq.impl.DefaultBinding$DefaultBytesBinding.get0(DefaultBinding.java:2192)
	at org.jooq.impl.DefaultBinding$DefaultJSONBBinding.get0(DefaultBinding.java:5575)
	at org.jooq.impl.DefaultBinding$DefaultJSONBBinding.get0(DefaultBinding.java:1)
	at org.jooq.impl.DefaultBinding$InternalBinding.get(DefaultBinding.java:1033)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1581)
	... 42 more
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.7 (https://github.com/jOOQ/jOOQ/issues/14467), and 3.16.13 (https://github.com/jOOQ/jOOQ/issues/14468)

**Comment by lukaseder** [comment]
`JSONB_OBJECTAGG` is also affected
